Freshli/IPackageRepository.cs

```diff
@@ -2,8 +2,12 @@
 
 namespace Freshli {
   public interface IPackageRepository {
-    VersionInfo LatestAsOf(DateTime date, string name);
-    VersionInfo VersionInfo(string name, string version);
-    VersionInfo Latest(string name, string thatMatches, DateTime asOf);
+    IVersionInfo VersionInfo(string name, string version);
+    IVersionInfo LatestAsOf(string name, DateTime asOf);
+    IVersionInfo LatestAsOfThatMatches(
+      string name,
+      DateTime asOf,
+      string thatMatches
+    );
```

Comment:
I reordered the parameters for improved consistency across all the methods.

Reply:
That's an improvement. I think this would be even better.

```
    IVersionInfo Latest(string name, DateTime asOf);
    IVersionInfo Latest(string name, DateTime asOf, string thatMatches);
```

And then the callers would look more like this:
```
Latest("lodash", asOf: date);
// and
Latest("lodash", asOf: date, thatMatches: "~1.2.3");
```

We can make that part of a separate pull request if it makes more sense.

Reply:
Updated in https://github.com/corgibytes/freshli/pull/54/commits/9c028eb6df1e7f0dec1b60b50af63020dff2e8e8.